include/boost/numeric/ublas/tensor/fixed_rank_tensor.hpp

```diff
@@ -0,0 +1,261 @@
+                "boost::numeric::ublas::fixed_rank_tensor( OtherTensor&& ) : The tensor should have same value type"
+            );
+
+            if ( other.extents().size() == R ){
+                throw std::runtime_error("boost::numeric::ublas::static_tensor( OtherTensor&& ) : The tensor should have equal number of extents");
+            }
+            
+            for( auto i = size_type(0); i < other.size(); ++i ){
+                super_type::data_[i] = std::move(other[i]);
+            }
+        }
+        
+        inline constexpr fixed_rank_tensor& operator=(const_reference v) noexcept{
+            std::fill(super_type::begin(), super_type::end(), v);
+            return *this;
+        }
+    };
+
+    template<typename V, typename E>
+    fixed_rank_tensor( E const&, V const& ) -> fixed_rank_tensor<V, E::_size, first_order>;
+    
+    template<typename E>
+    fixed_rank_tensor( E const& ) -> fixed_rank_tensor<float, E::_size, first_order>;
+
+} // boost::numeric::ublas
+
+
+namespace boost::numeric::ublas::detail{
+    
+    template<typename T, std::size_t R, typename F>
+    struct tensor_traits< fixed_rank_tensor<T, R, F> > {
+        using container_type= std::vector< T >;
+        using extents_type  = dynamic_extents<R>;
+        using layout_type   = F;
+        using container_tag = dynamic_tensor_tag;
+    };
+
+    template<typename T, std::size_t R, typename F>
+    struct is_tensor< fixed_rank_tensor<T, R, F> > : std::true_type{};
+    
+    template<typename T, std::size_t R, typename F, typename NewValue>
+    struct tensor_rebind< fixed_rank_tensor<T, R, F>, NewValue > {
+        using type = fixed_rank_tensor< NewValue, R, F >;
+    };
+    
+    template<typename T, std::size_t R, typename F, typename NewValue, typename NewExtents>
+    struct tensor_rebind< fixed_rank_tensor<T, R, F>, NewValue, NewExtents > {
+        using type = fixed_rank_tensor< NewValue, NewExtents::_size, F >;
+    };
+    
+    template<typename T, std::size_t R, typename F, typename NewValue, typename NewExtents, typename NewLayout>
+    struct tensor_rebind< fixed_rank_tensor<T, R, F>, NewValue, NewExtents, NewLayout > {
+        using type = fixed_rank_tensor< NewValue, NewExtents::_size, NewLayout >;
+    };
+
+    template<typename T, std::size_t R, typename F>
+    struct is_static< fixed_rank_tensor<T, R, F> > : std::false_type{};
+    
+    template<typename T, std::size_t R, typename F>
+    struct is_static_rank< fixed_rank_tensor<T, R, F> > : std::true_type{};
+    
+    template<typename T, std::size_t R, typename F>
+    struct is_dynamic< fixed_rank_tensor<T, R, F> > : std::true_type{};
+    
+    template<typename T, std::size_t R, typename F>
+    struct is_dynamic_rank< fixed_rank_tensor<T, R, F> > : std::false_type{};
+
+    template<typename V, typename F, typename T, std::size_t R>
+    struct result_tensor< V, basic_fixed_rank_extents<T,R>, F >{
+        using type = fixed_rank_tensor< V, R, F >;
+    };
```

Comment:
Could this be placed in a different file with all tensor traits ?
Also I would like to see it in `boost::numeric::ublas` not in detail namespace.

Reply:
This style allows us to easily remove and add *.hpp or makes code more module so if you do  not include then you do not need to know about it.

Reply:
Ok.